src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -65,33 +65,58 @@ std::vector<std::string> get_supported_metrics() {
           "confusion_matrix", "roc_curve"};
 }
 
-std::tuple<float,size_t> get_accuracy(size_t prediction_window, size_t num_classes, shared_float_array output_chunk, shared_float_array classes_chunk, data_iterator::batch::chunk_info info ){
+std::tuple<float,size_t> get_accuracy_per_sequence(size_t prediction_window, size_t num_classes, 
+  const shared_float_array& output_chunk, const shared_float_array& classes_chunk, data_iterator::batch::chunk_info info) {
       
 
-    const float* output_ptr = output_chunk.data();
-    const float* truth_ptr = classes_chunk.data();
+  const float* output_ptr = output_chunk.data();
+  const float* truth_ptr = classes_chunk.data();
     
-    size_t cumulative_samples = 0;
-    size_t seq_len = 0;
-    float cumulative_seq_accuracy = 0.f;
+  size_t cumulative_samples = 0;
+  size_t num_predictions = 0;
+  size_t num_correct_predictions = 0.f;
 
 
-      while (cumulative_samples < info.num_samples) {
-        cumulative_samples += prediction_window;
-       
-          if ((std::max_element(output_ptr, output_ptr + num_classes)- output_ptr == *truth_ptr)){
-            cumulative_seq_accuracy += 1.0;
-          };
-        seq_len++;
-        truth_ptr++;
-        output_ptr += num_classes;
+  while (cumulative_samples < info.num_samples) {
 
-      }
+    size_t prediction = std::max_element(output_ptr, output_ptr + num_classes) - output_ptr;
+
+    if (prediction == *truth_ptr) {
+      num_correct_predictions += 1;
+      };
+
+    cumulative_samples += prediction_window;
+    num_predictions++;
+    truth_ptr++;
+    output_ptr += num_classes;
+
+  }
+
+  return std::make_tuple(num_correct_predictions, num_predictions);
+
+}
+
+
+float get_accuracy_per_batch(size_t prediction_window, size_t num_classes, 
+  const shared_float_array& output, const data_iterator::batch& batch) {
+
+  float cumulative_per_batch_accuracy = 0.f;
 
-  return std::make_tuple(cumulative_seq_accuracy,seq_len);
+  for (size_t i = 0; i < batch.batch_info.size(); ++i){
 
+    const shared_float_array& output_chunk = output[i];
+    const shared_float_array& classes_chunk = batch.labels[i];
+    data_iterator::batch::chunk_info info = batch.batch_info[i];
+    size_t num_correct_predictions;
+    size_t num_predictions;
+
+    std::tie (num_correct_predictions, num_predictions) = get_accuracy_per_sequence(prediction_window, num_classes, output_chunk, classes_chunk, info);
```

Comment:
Nit: let's not leave a space between a function name `std::tie` and the opening `(` of its argument list